typed/remove_test.go

```diff
@@ -0,0 +1,133 @@
+package typed
+
+import (
+	"fmt"
+	"testing"
+
+	"sigs.k8s.io/structured-merge-diff/v3/fieldpath"
+	"sigs.k8s.io/structured-merge-diff/v3/value"
+)
+
+func TestRemove(t *testing.T) {
+
+	var cases = []struct {
+		object YAMLObject
+		remove *fieldpath.Set
+		deep   bool
+		expect YAMLObject
+	}{
+		{
+			object: `{}`,
+			remove: fieldpath.NewSet(fieldpath.MakePathOrDie("a")),
+			deep:   true,
+			expect: ``,
+		},
+		{
+			object: `{"a": "value"}`,
+			remove: fieldpath.NewSet(fieldpath.MakePathOrDie("a")),
+			deep:   true,
+			expect: `null`,
```

Comment:
I guess it's a little odd how `{}` goes to `` and this one goes to `null`. Do you happen to know why?

Reply:
I think it has something to do with how the test creates the typedvalues.
So the input gets converted as expected, after removal it results in {}

Maybe the result is not null, but null converts to the same as the result of an empty object?

Not sure if the way I create the objects (input, expect) is correct.

Reply:
I'll step through it a little more in the debugger to find out.

Reply:
Yeah, I commented below, I think the test code is fine, the tested code is probably not.

Reply:
Okay, so i found the cause.
https://github.com/kubernetes-sigs/structured-merge-diff/pull/133/files#diff-11232f9aff4ebc184b767f3a3520214aR78
There we just return when the map (same for list) is nil or empty.
This means the resulting object after remove is nil and the expect `` matches this.

Reply:
Not sure about the null one yet.